bin/inspect_all_known_projects.py

```diff
@@ -15,7 +15,7 @@
 DIR = Path(__file__).parent.resolve()
 
 
-def parse(contents: str) -> Optional[str]:
+def parse(contents: str) -> str | None:
```

Comment:
Why is it only rewriting things here? There's plenty of other use of `Optional[Foo]`?

Reply:
Because this file has `from __future__ import annotations` (a Python 3.7+ think). Our bin folder is Python 3.7+, though it must be picking it up from the future import rather than waiting for `--py37-plus`.

Reply:
Aaaaah, right, OK :-)

Reply:
If we require Python 3.7 for `bin` anyway, can we also have `--py37-plus` for this subfolder?

Reply:
> If we require Python 3.7 for `bin` anyway, can we also have `--py37-plus` for this subfolder?

That won't actually make a difference in this case, though: https://github.com/asottile/pyupgrade#pep-604-typing-rewrites